purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -614,6 +614,58 @@ class Purchases @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE) intern
         } ?: retrievePurchaseInfo(identityManager.currentAppUserID, listener)
     }
 
+    /**
+     * This function will change the current appUserID.
+     * Typically this would be used after a log out to identify a new user without calling configure
+     * @param newAppUserID The new appUserID that should be linked to the currently user
+     * @param [listener] An optional listener to listen for successes or errors.
+     */
+    @JvmOverloads
+    internal fun logIn(
+        newAppUserID: String,
+        listener: LogInListener? = null
+    ) {
+        identityManager.currentAppUserID.takeUnless { it == newAppUserID }?.let {
+            identityManager.logIn(newAppUserID,
+                    onSuccess = { purchaserInfo, created ->
+                        dispatch {
+                            listener?.onReceived(purchaserInfo, created)
+                            sendUpdatedPurchaserInfoToDelegateIfChanged(purchaserInfo)
+                        }
+                        fetchAndCacheOfferings(newAppUserID, state.appInBackground)
+                    },
+                    onError = { error ->
+                        dispatch { listener?.onError(error) }
+                    })
+        }
+                ?: retrievePurchaseInfo(identityManager.currentAppUserID, receivePurchaserInfoListener(
+                        onSuccess = { purchaserInfo ->
+                            dispatch { listener?.onReceived(purchaserInfo, false) }
+                        },
+                        onError = { error ->
+                            dispatch { listener?.onError(error) }
+                        }
+                ))
+    }
```

Comment:
I would apply formatting to this since the indentation looks off too. I also suggest a little trick, but running detekt as a pre-commit hook can fix some of this automatically I think.

```
#!/usr/bin/env bash
echo "Running detekt check..."
OUTPUT="/tmp/detekt-$(date +%s)"
./gradlew detektAll > $OUTPUT
EXIT_CODE=$?
if [ $EXIT_CODE -ne 0 ]; then
  cat $OUTPUT
  rm $OUTPUT
  echo "***********************************************"
  echo "                 Detekt failed                 "
  echo " Please fix the above issues before committing "
  echo "***********************************************"
  exit $EXIT_CODE
fi
rm $OUTPUT
```

Save that as a pre-commit file in your .git/hooks and give it executable permissions.